sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereLogic.cs

```diff
@@ -135,53 +142,54 @@ public WhereLogic IsOfModel(string model, bool exact = false)
             }
 
             whereClauseArg.Append(')');
-            _clauses.Add(new WhereClause(whereClauseArg.ToString()));
+            _conditions.Append(whereClauseArg.ToString());
 
-            return this;
+            return _logicalOperator;
         }
 
         /// <summary>
         /// Adds a user-specified ADT query language function to check an expression's type against a built type in the ADT query language.
         /// </summary>
         /// <param name="expression"> The expression that the query is looking for as a specified type. </param>
         /// <param name="type"> The type in the ADT query language being checked for. </param>
-        /// <returns></returns>
-        public WhereLogic IsOfType(string expression, AdtDataType type)
+        /// <returns> Logical operator to combine different WHERE functions or conditions. </returns>
+        public LogicalOperator IsOfType(string expression, AdtDataType type)
+        {
+            string functionName = QueryConstants.DataTypeToFunctionNameMap[type];
+            _conditions.Append(new WhereClause($"{functionName}({expression})").Condition);
+            return _logicalOperator;
+        }
+
+        /// <summary>
+        /// Used to add nested WHERE conditions to a query.
+        /// </summary>
+        /// <param name="nested"> WhereLogic methods to perform within a set of parenthesis. </param>
+        /// <returns> Logical operator to combine different WHERE functions or conditions. </returns>
+        public LogicalOperator IsTrue(Func<WhereLogic, LogicalOperator> nested)
         {
-            string functionName = QueryConstants.IsOfTypeConversions[type];
-            _clauses.Add(new WhereClause($"{functionName}({expression})"));
-            return this;
+            var nestedLogic = new WhereLogic(null);
+            nested.Invoke(nestedLogic);
+
+            _conditions.Append(new WhereClause(QueryConstants.OpenParenthesis + nestedLogic.GetQueryText() + QueryConstants.CloseParenthesis).Condition);
+
+            return _logicalOperator;
         }
 
         /// <inheritdoc/>
-        public override AdtQueryBuilder Build()
+        internal AdtQueryBuilder Build()
```

Comment:
We can even rename this method to clearly distinguish the QueryBase Build method and one-off methods that look like it.

Reply:
same for GetQueryText

Reply:
`BuildLogic()`/

Reply:
`GetLogicText()`?

Reply:
Sure